packages/pigeon/lib/objc_generator.dart

```diff
@@ -953,47 +951,61 @@ static id GetNullableObjectAtIndex(NSArray* array, NSInteger key) {
       });
     }
 
-    void writeFromMap() {
-      indent.write('+ ($className *)fromMap:(NSDictionary *)dict ');
+    void writeFromList() {
+      indent.write('+ ($className *)fromList:(NSMutableArray *)list ');
       indent.scoped('{', '}', () {
         const String resultName = 'pigeonResult';
         indent.writeln('$className *$resultName = [[$className alloc] init];');
-        for (final NamedType field in klass.fields) {
+        final int len = klass.fields.length;
+        indent.scoped('if (list == nil || list == [NSNull null]) {', '}', () {
+          indent.writeln(
+              'list = [[NSMutableArray alloc] initWithCapacity: $len];');
+          indent.write('for (int i = 0; i < $len; i++) ');
+          indent.scoped('{', '}', () {
+            indent.writeln('[list addObject: [NSNull null]];');
+          });
+        });
+        klass.fields
+            .toList()
+            .asMap()
+            .forEach((int index, final NamedType field) {
           if (enumNames.contains(field.type.baseName)) {
             indent.writeln(
-                '$resultName.${field.name} = [${_dictGetter(classNames, 'dict', field, options.prefix)} integerValue];');
+                '$resultName.${field.name} = [(${_listGetter(classNames, 'list', field, index, options.prefix)} == [NSNull null] ? 0 : ${_listGetter(classNames, 'list', field, index, options.prefix)}) integerValue];');
```

Comment:
I haven't dug into all the usage, but can `_listGetter` convert `NSNull` to `nil` internally? If so we should do that, and if not we should make a second trivial wrapper that does, and use that here. Then this double-lookup and conditional logic goes away, since`[nil integerValue]` is 0.

Reply:
This is an oversight on my part, I just need to use the getNullableObjectAtIndex method.